src/sagemaker/jumpstart/curated_hub/accessors/curated_hub_s3_accessor.py

```diff
@@ -0,0 +1,102 @@
+
+
+class CuratedHubS3Accessor(JumpstartS3Accessor):
+    """Helper class to access Curated Hub s3 bucket"""
+
+    def __init__(self, region: str, bucket: str):
+        self._region = region
+        self._bucket = bucket
+        self._studio_metadata_map = get_studio_model_metadata_map_from_region(
+            region
+        )  # Necessary for SDK - Studio metadata drift
+
+    def get_bucket(self) -> str:
+        """Retrieves s3 bucket"""
+        return self._bucket
+
+    def get_inference_artifact_s3_reference(
+        self, model_specs: JumpStartModelSpecs
+    ) -> S3ObjectReference:
+        """Retrieves s3 reference for model inference artifact"""
+        return create_s3_object_reference_from_bucket_and_key(
+            self.get_bucket(), f"{model_specs.model_id}/{model_specs.version}/infer.tar.gz"
+        )
+
+    def get_inference_script_s3_reference(
+        self, model_specs: JumpStartModelSpecs
+    ) -> S3ObjectReference:
+        """Retrieves s3 reference for model traiing script"""
+        return create_s3_object_reference_from_bucket_and_key(
+            self.get_bucket(), f"{model_specs.model_id}/{model_specs.version}/sourcedir.tar.gz"
+        )
+
+    def get_training_artifact_s3_reference(
+        self, model_specs: JumpStartModelSpecs
+    ) -> S3ObjectReference:
+        """Retrieves s3 reference for model training artifact"""
+        return create_s3_object_reference_from_bucket_and_key(
+            self.get_bucket(), f"{model_specs.model_id}/{model_specs.version}/train.tar.gz"
+        )
+
+    def get_training_script_s3_reference(
+        self, model_specs: JumpStartModelSpecs
+    ) -> S3ObjectReference:
+        """Retrieves s3 reference for model training script"""
+        return create_s3_object_reference_from_bucket_and_key(
+            self.get_bucket(),
+            f"{model_specs.model_id}/{model_specs.version}/training/sourcedir.tar.gz",
+        )
+
+    def get_demo_notebook_s3_reference(self, model_specs: JumpStartModelSpecs) -> S3ObjectReference:
+        """Retrieves s3 reference for model jupyter notebook"""
+        return create_s3_object_reference_from_bucket_and_key(
+            self.get_bucket(), f"{model_specs.model_id}/{model_specs.version}/demo-notebook.ipynb"
+        )
+
+    def get_default_training_dataset_s3_reference(
+        self, model_specs: JumpStartModelSpecs
+    ) -> S3ObjectReference:
+        """Retrieves s3 reference for s3 directory containing training datasets"""
+        return create_s3_object_reference_from_bucket_and_key(
+            self.get_bucket(), self._get_training_dataset_prefix(model_specs)
+        )
+
+    def _get_training_dataset_prefix(
+        self, model_specs: JumpStartModelSpecs
+    ) -> str:  # Studio expects the same format as public hub bucket
+        studio_model_metadata = self._studio_metadata_map[model_specs.model_id]  # TODO: verify this
```

Comment:
this is dangerous since there's a coupling now with studio metadata. we should put everything we need in the same metadata